tests/network-tests/src/fixtures/referendum/voteFixture.ts

```diff
@@ -0,0 +1,68 @@
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import { ISubmittableResult } from '@polkadot/types/types'
+import BN from 'bn.js'
+import { assert } from 'chai'
+import { Api } from '../../Api'
+import { StandardizedFixture } from '../../Fixture'
+import { QueryNodeApi } from '../../QueryNodeApi'
+import { EventDetails, AnyQueryNodeEvent } from '../../types'
+
+export type VotingParams = {
+  commitment: string
+  stake: BN
+}
+
+export class VoteFixture extends StandardizedFixture {
+  private votes: Map<string, VotingParams>
+  private failureExpected: boolean
+
+  constructor(api: Api, query: QueryNodeApi, votes: Map<string, VotingParams>, failureExpected = false) {
+    super(api, query)
+    this.votes = votes
+    this.failureExpected = failureExpected
+  }
+
+  protected getSignerAccountOrAccounts(): Promise<string | string[]> {
+    return Promise.resolve(Array.from(this.votes.keys()))
+  }
+
+  protected getExtrinsics(): Promise<
+    SubmittableExtrinsic<'promise', ISubmittableResult>[] | SubmittableExtrinsic<'promise', ISubmittableResult>[][]
+  > {
+    const results = Array.from(this.votes).map(async ([, params]) =>
+      this.api.tx.referendum.vote(params.commitment, params.stake)
+    )
+    return Promise.all(results)
+  }
+
+  protected getEventFromResult(result: ISubmittableResult): Promise<EventDetails<unknown>> {
+    return this.api.getEventDetails(result, 'referendum', 'VoteCast')
+  }
+
+  protected assertQueryNodeEventIsValid(qEvent: AnyQueryNodeEvent, i: number): void {
+    // TODO: implement QN checks
+  }
+
+  // overwriting for using decremental tips
+  public async execute(): Promise<void> {
+    const accountOrAccounts = await this.getSignerAccountOrAccounts()
+    const extrinsics = await this.getExtrinsics()
+    this.extrinsics = super.flattenExtrinsics(extrinsics)
+    await this.api.prepareAccountsForFeeExpenses(accountOrAccounts, this.extrinsics, 10)
+    this.results = await this.api.sendExtrinsicsAndGetResults(extrinsics, accountOrAccounts, 10)
+
+    if (!this.failureExpected) {
+      this.events = await Promise.all(this.results.map((r) => this.getEventFromResult(r)))
+    } else {
+      if (Array.isArray(accountOrAccounts)) {
```

Comment:
I think we should be looking for event `AccountAlreadyOptedOutOfVoting` to be sure it is failing for the right reason. It could fail for other reason, incase the account setup in the fixture was incorrect for example.